tests/s3_bucket/test_github_content_media.py

```diff
@@ -0,0 +1,39 @@
+import requests
+
+"""
+Verifies that media folder in every github content repo exists and is not empty.
+Latest update on September 16th, 2021
+"""
+
+
+def test_github_content_media(git_content_repos, headers_data):
+
+    html_url_count = []
+
+    for repo in git_content_repos:
+
+        print("\nNow verifying: ", repo)
+
+        media_dir = f"https://api.github.com/repos/openstax/{repo}/contents/media/"
+
+        media_list = requests.get(media_dir, headers=headers_data)
+
+        if media_list.status_code != 200:
+
+            # Return code 404, 501, ... for incorrect media url
+            print(f">>>>> FAILED {media_list.status_code}: no media folder in {repo}")
+
+        else:
+
+            for item in media_list.json():
+
+                if item["type"] != "file":
+
+                    # Ignore anything that may not be a file
+                    continue
+
+                else:
+
+                    html_url_count.append(item["html_url"])
+
+        assert sum(".jpg" in s for s in html_url_count) > 0
```

Comment:
Question / thought: I believe this assumes a valid media directory will have a `*.jpg` file in it. However: 1) There might be cases for new books early in development where there are some media files but they don't happen to be of that extension; 2) I think there early on it's valid for the media directory to only have a `.gitkeep` file (maybe worth asking / confirming with the team). In both cases this assertion may lead to false positive failures.

Reply:
Thanks. You are right. Maybe I should check for any kind of file in the folder or just that the folder exists... but I would like to know if there is something in it as well in case there were changes and the media files were deleted or not copied back or left out completely... Yeah, the development phase might be a problematic phase...